core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -172,15 +179,24 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             // we'll set a cookie so the subsequent operations succeed
             if (request instanceof HttpServletRequest) {
                 HttpServletRequest req = (HttpServletRequest)request;
-                //if (!Pattern.compile(".*[.](css|ttf|gif|woff|eot|png|js)").matcher(req.getRequestURI()).matches()) {
-                    // Allow js & css requests through
-                if((req.getContextPath() + "/").equals(req.getRequestURI())) {
+                // Allow js & css requests through
+                if (!RESOURCE_PATTERN.matcher(req.getRequestURI()).matches()) { // allow resources
+                    boolean isAdmin = jenkins.getACL().hasPermission(Jenkins.ADMINISTER);
+                    final String adjustedPath = req.getRequestURI().substring(req.getContextPath().length());
+                    final String destination;
+                    // if the user hasn't authenticated, don't allow anything other than the login submit and error page
+                    if (!isAdmin && !ALLOWED_PATHS.matcher(adjustedPath).matches()) {
```

Comment:
This logic does not comply with the comment above. If we start providing new users during startup (e.g. useful for the DEV environment), then it will redirect non-admin users to the login page.

I think it makes sense to explicitly check the non-authenticated use-case